include/swift/SILOptimizer/Utils/InstOptUtils.h

```diff
@@ -289,7 +289,9 @@ void insertDestroyOfCapturedArguments(
     SILLocation loc = RegularLocation::getAutoGeneratedLocation());
 
 void insertDeallocOfCapturedArguments(PartialApplyInst *pai,
-                                      DominanceInfo *domInfo);
+    DominanceInfo *domInfo,
+    llvm::function_ref<bool(SILValue)> shouldInsertDestroy =
```

Comment:
IIRC, you can do this more efficiently by passing in a nullptr as the default parameter for llvm::function_ref and just check if function_ref is nullptr before calling it. This means in the common case, we have a well predicated conditional instead of calling an indirect pointer.

My memory may be incorrect though.

Reply:
`insertDestroyOfCapturedArguments` doesn't do that. Using `nullptr` also makes calling the function more mistake-prone since code has to check for null at every use site. A modern CPU is going to predict an indirect branch that always goes the same place about as well as a binary branch that does the same so I don't think a minor CPU resource optimization is really worth the added maintenance hazard.